### PR TITLE
refactor!: poetry, github action overhaul, pyenv / asdf versions

### LINKED ISSUE #136: Test infrastructure, packaging, etc. needs overhaul
Here is all a user should need to bootstrap themselves:

```console
poetry install
```

And to test:

```console
poetry run py.test
```

It seems that `tox` is set as a _requirement_, not a supplement, to testing on the django-robots - blocking any hope of accomplishing even basic tasks. 

CI could use test matrix to get django

This is needed for #133 and for future maintainers #135

### PR DESCRIPTION
**Pull request by tony**
Fixes #136

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,48 +1,80 @@
...
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        run: poetry build
+
+      - name: Publish package
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
+          skip_existing: true
```
Comment (jezdez) [inline .github/workflows/test.yml:80]: This should not be done here but in the release.yml file instead. Please don't push to PyPI directly, too.

File: .tmuxp.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+session_name: django-robots
```
Comment (jezdez) [inline .tmuxp.yaml:1]: That seems not needed here.

File: .tool-versions [hunk 3]
```diff
@@ -0,0 +1,2 @@
+poetry 1.2.1
```
Comment (jezdez) [inline .tool-versions:1]: This isn't needed for development should be kept on your machine.

**Comment by tony** [comment]
@jezdez I meant to close this PR.  This was just experimenting ^

**Comment by tony** [comment]
>  I'd rather switch to hatch than poetry to be honest, as it's probably better supported via the PyPA. 

[hatch](https://pypi.org/project/hatch/), that's totally new to me!

> I like seeing the move from setup.py to pyproject.toml though!

I'm happy to contribute that in a separate PR (and even try hatch out)

That said - the bigger issue I'm concerned about is #124.

@jezdez What are your thoughts on this project here and now? Am I doing more harm than good? Do you want to pilot it again? (I notice this **specific django package** is your creation, on atop of your being apart of jazzband).  Can / should I email you?

**Comment by tony** [comment]
For now I will close this - per https://github.com/jazzband/django-robots/pull/137#issuecomment-1280049961

File: .github/workflows/test.yml (hunk 1 above)
Comment (tony) [reply .github/workflows/test.yml:80]: That's true ☝️ And I know why that'd be. This is copied from my own project's workflows (it wasn't meant to be added)

**Comment by tony** [comment]
@jezdez 

Where I took this:

1. re: hatch, pipenv, poetry, etc: https://github.com/jazzband/help/issues/313
2. Best practices: https://github.com/jazzband/help/issues/314

   in the above I also floated the idea of:
   - a "flagship" project that can champion the best practices for other projects to imitate
   - potential sprints to update best practices by hand (e.g. setup.py -> pyproject.toml across all projects)

**Comment by ofek** [comment]
> potential sprints to update best practices by hand (e.g. setup.py -> pyproject.toml across all projects)

fyi Hatch can pretty robustly automate that https://hatch.pypa.io/latest/intro/#existing-project

**Comment by ofek** [comment]
I did one 🙂 https://github.com/jazzband/django-debug-toolbar/pull/1690

**Pull request closed without merging**